### DISCUSSION TITLE
Livewire component rendering data slowly through model update?

### DISCUSSION THREAD

**Discussion by jackpipergit** [livewire/livewire#6975] [1. Help] [answered]
Hi all,

I've been experimenting with Livewire 3 to see if it suits my side project for work. I'm pleased that there are no issues with data rendering upon initial page load or reload. However, as soon as I update through wire:model.live, the data takes an extraordinarily long time to load. The query returns 450 results, with not many columns – only three, the longest of which has 30 characters.

In the last couple of tests I've run, the initial page load (with data fully loaded via the render method) took less than 350ms. Yet, after attempting to update the $selectedWarehouse, which triggers the data to reload, it takes between 14,000ms and 20,000ms at times.

I've even simplified the query down to its most basic form, targeting just one table with no relationships, but I'm seeing the same result.

Apologies if it's something simple; I'm just a hobbyist, but I can't seem to get my head around this issue. To me, if it loads quickly upon initial render, why is it taking so long to reload?

Any help would be much appreciated! - running the latest stable Laravel & Livewire

Livewire component
```
<?php

namespace App\Livewire\Reports;

use App\Models\SAPB1\Inventory\OITM;
use App\Models\SAPB1\Inventory\OWHS;
use Livewire\Component;

class WeeksOfStock extends Component
{
    public $items;
    public $warehouses;
    public $selectedWarehouse;

    public function mount()
    {
        $this->warehouses = OWHS::select('WhsCode', 'WhsName')->get(); // Tried replacing this with a static array/object but same slow results.
        $this->selectedWarehouse = 'all';
        $this->loadData();
    }

    public function loadData()
    {
        $this->items = OITM::select('ItemCode', 'ItemName', 'OnHand')
            ->where([
                ['InvntItem', '=', 'Y'],
                ['ItmsGrpCod', '<>', 111],
                ['frozenFor', '=', 'N'],
                ['ItemName', 'not like', '%sample%']
            ])->get();
    }

    public function updatedSelectedWarehouse()
    {
        $this->loadData();
    }

    public function render()
    {
        return view('livewire.reports.weeks-of-stock');
    }
}


```

Blade Component that triggers update to updatedSelectedWarehouse()
```
    <div>
        <select wire:model.live="selectedWarehouse">
            <option value="all">All Warehouses</option>
            @foreach($warehouses as $warehouse)
                <option value="{{ $warehouse->WhsCode }}">{{ $warehouse->WhsName }}</option>
            @endforeach
        </select>
    </div>
```

**Comment by jackpipergit** [comment, marked as answer]
Managed to fix the issues from a resolution suggested by Snapey over on Laracasts, 

https://laracasts.com/discuss/channels/livewire/livewire-component-rendering-data-slowly-through-model-update?page=1&replyId=907994

**Reply by PhiloNL** [reply]
Check out the computed properties documentation as well. This has additional features like caching data.
